### PR TITLE
Fix ICS export & update metadata

### LINKED ISSUE #125: Bug — Export to ICS
**Describe the bug**
1. Export seems to create 2 events during the same day
Screen Shot 2020-04-24 at 1 58 35 PM
2. The event is not repeated through expected date range

**To Reproduce**
1. Add a "Data Mining" from Fall 2020 to schedule
2. Export to ICS
3. Use calendar import ICS functionality to pull in file
4. See calendar

**Expected behavior**
- See the class repeated multiple times throughout date range (use repeat) [see bottom for more info `RRULE:FREQ=WEEKLY;BYDAY=TU,FR;UNTIL=20201221T150000Z`]
- Name of event should say "Data Mining" using the new full title
- No location data for now (remove "null")
> Screen Shot 2020-04-24 at 2 02 54 PM

**Desktop (please complete the following information):**
- MacOS + Apple Calendar

**Additional context**
I used this [tool](https://ical.marudot.com/#) to generate the ics file attached that should be the expected ICS file

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//ical.marudot.com//iCal Event Maker
CALSCALE:GREGORIAN
BEGIN:VTIMEZONE
TZID:America/New_York
TZURL:http://tzurl.org/zoneinfo-outlook/America/New_York
X-LIC-LOCATION:America/New_York
BEGIN:DAYLIGHT
TZOFFSETFROM:-0500
TZOFFSETTO:-0400
TZNAME:EDT
DTSTART:19700308T020000
RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=2SU
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:-0400
TZOFFSETTO:-0500
TZNAME:EST
DTSTART:19701101T020000
RRULE:FREQ=YEARLY;BYMONTH=11;BYDAY=1SU
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
DTSTAMP:20200424T181448Z
UID:20200424T181448Z-2088954976@marudot.com
DTSTART;TZID=America/New_York:20200828T100000
RRULE:FREQ=WEEKLY;BYDAY=TU,FR;UNTIL=20201221T150000Z
DTEND;TZID=America/New_York:20200828T115000
SUMMARY:Data Mining
END:VEVENT
END:VCALENDAR
```

### PR DESCRIPTION
**Pull request by CoreyBuckley**
**Issue**

Closes #125 

**Test Procedure**

Show procedure to test functionality with a clear procedure

1. Build & run the project. Upload Fall 2020 semester if not already there.
2. On schedule page, add class Data Mining `CSCI-4390`
3. Export to ICS (button is near the footer)
4. Look at in your favorite calendar app and one event for tues & fri should appear weekly @ 10am-11:50am until the week of 12/21.
5. The event information should show 'Graph Mining' as the class and the location should be blank.

**Photos**

N/A

**Additional Info**

Some clients care about capitalization of ICS rules. Problem was b/c `rrule` should have been capitalized. Haha.

(Credit to @jshom for finding that weird bug)

### CODE REVIEW

File: src/web/public/ics.min.js [hunk 1]
```diff
@@ -23,6 +23,6 @@ SOFTWARE.
...
...ring()).slice(-2),F=("00"+n.getSeconds().toString()).slice(-2),G="",H="";r+s+t+x+y+z!=0&&(G="T"+r+s+t,H="T"+x+y+z);var I,J="T"+D+E+F,K=o+p+q+G,L=u+v+w+H,M=A+B+C+J;if(j)if(j.rrule)I=j.rrule;else{if(I="RRULE:FREQ="+j.freq,j.until){var N=new Date(Date.parse(j.until)).toISOString();I+=";UNTIL="+N.substring(0,N.length-13).replace(/[-]/g,"")+"000000Z"}j.interval&&(I+=";INTERVAL="+j.interval),j.count&&(I+=";COUNT="+j.count),j.byday&&j.byday.length>0&&(I+=";BYDAY="+j.byday.join(","))}var O=((new Date).toISOString(),["BEGIN:VEVENT","UID:"+d.length+"@"+a,"CLASS:PUBLIC","DESCRIPTION:"+e,"DTSTAMP;VALUE=DATE-TIME:"+M,"DTSTART;VALUE=DATE-TIME:"+K,"DTEND;VALUE=DATE-TIME:"+L,"LOCATION:"+f,"SUMMARY;LANGUAGE=en-us:"+b,"TRANSP:TRANSPARENT","END:VEVENT"]);return I&&O.splice(4,0,I),O=O.join(c),d.push(O),O},download:function(a,b){if(d.length<1)return!1;b="undefined"!=typeof b?b:".ics",a="undefined"!=typeof a?a:"calendar";var g,h=e+c+d.join(c)+f;if(-1===navigator.userAgent.indexOf("MSIE 10"))g=new Blob([h]);else{var i=new BlobBuilder;i.append(h),g=i.getBlob("text/x-vCalendar;charset="+document.characterSet)}return saveAs(g,a+b),h},build:function(){if(d.length<1)return!1;var a=e+c+d.join(c)+f;return a}}};
```
Comment (123joshuawu) [inline src/web/public/ics.min.js:28]: ok i think we need some proper indentation, I can't read anything

File: src/web/src/pages/Main.vue [hunk 2]
```diff
@@ -290,9 +290,10 @@ export default {
             exclusive_date_end.setDate(course.date_end.getDate() + 1);
             semester = session.semester;
             calendarBuilder.addEvent(
-              `Class: ${course.title}`,
+              `Class: ${course.full_title ? course.full_title : course.title}`,
```
Comment (123joshuawu) [inline src/web/src/pages/Main.vue:293]: what about `course.full_title || course.title` :)

File: src/web/public/ics.min.js (hunk 1 above)
Comment (CoreyBuckley) [reply src/web/public/ics.min.js:28]: u meme'n rn?

File: src/web/src/pages/Main.vue (hunk 2 above)
Comment (CoreyBuckley) [reply src/web/src/pages/Main.vue:293]: that makes sense.

File: src/web/src/pages/Main.vue [hunk 3]
```diff
@@ -310,9 +310,10 @@ export default {
             exclusive_date_end.setDate(course.date_end.getDate() + 1);
             semester = session.semester;
             calendarBuilder.addEvent(
-              `Class: ${course.title}`,
+              `Class: ${course.full_title || course.title}`,
```
Comment (jshom) [inline src/web/src/pages/Main.vue:313]: Let's remove the Class prefix, given most of us are full time students, there's no need to specify context in the name of the event.

File: src/web/src/pages/Main.vue [hunk 4]
```diff
@@ -310,9 +310,10 @@ export default {
...
               'LEC day',
-              session.location,
+//              session.location,
```
Comment (marchdan) [inline src/web/src/pages/Main.vue:315]: If you're going to leave location present but commented, you should also have professor there was as well

Comment (CoreyBuckley) [reply src/web/src/pages/Main.vue:315]: location is its own special field in the ICS stuff. I can add a comment to indicate professor should be added to the description

**Comment by marchdan** [comment]
@jshom You have to approve this since you asked for changes

**Comment by jshom** [comment]
```
BEGIN:VCALENDAR
PRODID:Calendar
VERSION:2.0
BEGIN:VEVENT
UID:0@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200822T000000Z;INTERVAL=1;BYDAY=TH
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200526T103000
DTEND;VALUE=DATE-TIME:20200526T123500
LOCATION:
SUMMARY;LANGUAGE=en-us:Materials and Design
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
UID:1@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200711T000000Z;INTERVAL=1;BYDAY=TU
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200526T131500
DTEND;VALUE=DATE-TIME:20200526T155000
LOCATION:
SUMMARY;LANGUAGE=en-us:Introduction to Biology Laboratory
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
UID:2@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200711T000000Z;INTERVAL=1;BYDAY=FR
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200526T131500
DTEND;VALUE=DATE-TIME:20200526T155000
LOCATION:
SUMMARY;LANGUAGE=en-us:Introduction to Biology Laboratory
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
UID:3@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200822T000000Z;INTERVAL=1;BYDAY=TH
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200713T133000
DTEND;VALUE=DATE-TIME:20200713T153500
LOCATION:
SUMMARY;LANGUAGE=en-us:Documentary Film
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
UID:4@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200822T000000Z;INTERVAL=1;BYDAY=FR
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200713T133000
DTEND;VALUE=DATE-TIME:20200713T153500
LOCATION:
SUMMARY;LANGUAGE=en-us:Documentary Film
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
UID:5@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200822T000000Z;INTERVAL=1;BYDAY=MO
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200713T133000
DTEND;VALUE=DATE-TIME:20200713T153500
LOCATION:
SUMMARY;LANGUAGE=en-us:Documentary Film
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
UID:6@default
CLASS:PUBLIC
DESCRIPTION:
RRULE:FREQ=WEEKLY;UNTIL=20200822T000000Z;INTERVAL=1;BYDAY=TU
DTSTAMP;VALUE=DATE-TIME:20200424T181334
DTSTART;VALUE=DATE-TIME:20200713T133000
DTEND;VALUE=DATE-TIME:20200713T153500
LOCATION:
SUMMARY;LANGUAGE=en-us:Documentary Film
TRANSP:TRANSPARENT
END:VEVENT
END:VCALENDAR
```
With events that occur multiple times a week, a new vincent should not be made, one per class. otherwise on mac it's off
Screen Shot 2020-04-24 at 6 15 34 PM

**Pull request merged**